src/proxy/util.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::{Error, ErrorKind};
+
+pub fn is_runtime_shutdown(e: &Error) -> bool {
+    match e.kind() {
+        ErrorKind::Other => {
```

Comment:
I am not sure which is the idiotic way for rust

`match vs if`

Reply:
Since we care about both conditions, I'd go with `if ... && ...` instead of this `match` and `==` mix.

Reply:
ok